lib/internal/modules/cjs/loader.js

```diff
@@ -1293,5 +1294,15 @@ Module.syncBuiltinESMExports = function syncBuiltinESMExports() {
   }
 };
 
+Module.isBuiltIn = function isBuiltIn(moduleName) {
```

Comment:
I think that it would be better to throw when `moduleName` is not a string. then we could remove all `String(moduleName)` conversions.

just like what `is-builtin-module` do: https://github.com/sindresorhus/is-builtin-module/blob/819869f9a6b64dffc80ac5b39c26def695afaef4/index.js#L8-L10

```suggestion
Module.isBuiltIn = function isBuiltIn(moduleName) {
  validateString(moduleName, 'moduleName');
  
```

Reply:
`String` conversions are perf hits? Also, what is the advantage of throwing instead of just processing and returning a false? [I understand it would be more of a `noop` if the validation fails, but from the user perfective? As we make it clear that the `moduleName` must be a `string`? The doc does state that it must be a string.

Reply:
> `String` conversions are perf hits?

No, I guess I am wrong. After some thinking, I think an `isXXX` method throw error will be a little weird. I could not find a precedent in node code.

I will agree with you and make this function more forgiving.

Reply:
Nit: but it would be better if we do a `typeof moduleName !== 'string'` at the start of the method, If it is not true, then we return false earlier?

```js
if (typeof moduleName !== 'string') {
  return false;
}
```